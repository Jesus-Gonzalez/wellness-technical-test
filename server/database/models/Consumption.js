import mongoose from 'mongoose'

export const ConsumptionSchema = new mongoose.Schema({
  date: Date,
  consumption: Number,
  price: Number,
  cost: Number
})

export const ConsumptionModel = mongoose.model('Consumption', ConsumptionSchema)
