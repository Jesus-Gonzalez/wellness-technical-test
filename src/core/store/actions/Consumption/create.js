import * as ActionTypes from 'core/store/actionTypes'

export const createConsumption = item => ({
  type: ActionTypes.Consumption.Create,
  item
})
