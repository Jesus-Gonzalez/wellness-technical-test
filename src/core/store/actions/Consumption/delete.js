import * as ActionTypes from 'core/store/actionTypes'

export const deleteConsumption = item => ({
  type: ActionTypes.Consumption.Fetch,
  item
})
