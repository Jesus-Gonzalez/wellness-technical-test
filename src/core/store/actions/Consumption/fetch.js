import * as ActionTypes from 'core/store/actionTypes'

export const fetchConsumptions = () => ({
  type: ActionTypes.Consumption.Fetch
})
