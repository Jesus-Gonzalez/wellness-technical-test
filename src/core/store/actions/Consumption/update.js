import * as ActionTypes from 'core/store/actionTypes'

export const updateConsumptions = item => ({
  type: ActionTypes.Consumption.Fetch,
  item
})
