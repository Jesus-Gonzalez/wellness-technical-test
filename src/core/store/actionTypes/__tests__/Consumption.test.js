import { Consumption } from '../Consumption.enum'

describe('Actions', () => {
  it('Consumption should return the types object', () => {
    expect(Consumption).toMatchSnapshot()
  })
})
