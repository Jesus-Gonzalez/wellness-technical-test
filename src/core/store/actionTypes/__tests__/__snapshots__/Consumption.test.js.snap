// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions Consumption should return the types object 1`] = `
Object {
  "Create": "CREATE_CONSUMPTIONS",
  "CreateError": "CREATE_CONSUMPTIONS_ERROR",
  "CreateSuccess": "CREATE_CONSUMPTIONS_SUCCESS",
  "Delete": "DELETE_CONSUMPTIONS",
  "DeleteError": "DELETE_CONSUMPTIONS_ERROR",
  "DeleteSuccess": "DELETE_CONSUMPTIONS_SUCCESS",
  "Fetch": "FETCH_CONSUMPTIONS",
  "FetchError": "FETCH_CONSUMPTIONS_ERROR",
  "FetchSuccess": "FETCH_CONSUMPTIONS_SUCCESS",
  "Update": "UPDATE_CONSUMPTIONS",
  "UpdateError": "UPDATE_CONSUMPTIONS_ERROR",
  "UpdateSuccess": "UPDATE_CONSUMPTIONS_SUCCESS",
}
`;
